Source/DafnyLanguageServer/Caching/ByteArrayEquality.cs

```diff
@@ -0,0 +1,34 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Policy;
+
+namespace Microsoft.Dafny.LanguageServer.Language;
+
+class ByteArrayEquality : IEqualityComparer<byte[]> {
+  public int Compare(byte[]? x, byte[]? y) {
+    return (x, y) switch {
+      (null, null) => 0,
+      (null, _) => 1,
+      (_, null) => -1,
+      _ => x.SequenceEqual(y) ? 0 : 1
+    };
+  }
+
+  public bool Equals(byte[]? x, byte[]? y) {
+    return (x, y) switch {
+      (null, null) => true,
+      (null, _) => false,
+      (_, null) => false,
+      _ => x.SequenceEqual(y)
+    };
+    ;
+  }
+
+  public int GetHashCode(byte[] obj) {
+    if (obj.Length == 0) {
+      return 0;
+    }
+    return BitConverter.ToInt32(obj, 0);
+  }
```

Comment:
Can you add some documentation on how this hash code is being used? It seems unusual to use the first four bytes of a byte array as a "hash".